resources/volume.rb

```diff
@@ -27,16 +27,30 @@
   # @return [OneviewSDK::Volume] Loaded Volume resource
   def load_resource_with_associated_resources
     item = load_resource
-    item = load_storage_system(item)
+    load_storage_system(item)
 
     # item.set_storage_pool(OneviewSDK::StoragePool.new(item.client, name: storage_pool)) if storage_pool
-    # HACK Ruby SDK 1.0.0 issue workaround
-    sp = OneviewSDK::StoragePool.new(item.client, name: storage_pool)
-    raise "Storage Pool '#{sp['name']}' not found" unless sp.retrieve!
-    item['provisioningParameters']['storagePoolUri'] = sp['uri']
+    # Workaround for issue in oneview-sdk:
+    if storage_pool
+      sp = OneviewSDK::StoragePool.find_by(item.client, name: storage_pool).first
+      raise "Storage Pool '#{sp['name']}' not found" unless sp
+      item['storagePoolUri'] = sp['uri']
+    end
 
     item.set_snapshot_pool(OneviewSDK::StoragePool.new(item.client, name: snapshot_pool)) if snapshot_pool
-    item.set_volume_template(OneviewSDK::VolumeTemplate.new(item.client, name: volume_template)) if volume_template
+    item.set_storage_volume_template(OneviewSDK::VolumeTemplate.new(item.client, name: volume_template)) if volume_template
+
+    # Convert capacity integers to strings
+    item['provisionedCapacity'] = item['provisionedCapacity'].to_s if item['provisionedCapacity']
+    item['allocatedCapacity'] = item['allocatedCapacity'].to_s if item['allocatedCapacity']
+
+    unless item.exists? # Also set provisioningParameters if the volume does not exist
+      item['provisioningParameters'] ||= {}
+      item['provisioningParameters']['shareable'] = item['shareable'] if item['provisioningParameters']['shareable'].nil?
+      item['provisioningParameters']['provisionType'] ||= item['provisionType']
+      item['provisioningParameters']['requestedCapacity'] ||= item['provisionedCapacity']
```

Comment:
This section is just allowing you to set the capacity like you would if you did an update. `item['provisionedCapacity']` exists once the volume is created, but `item['requestedCapacity']` does not (it only exists in the create, and it's `item['provisioningParameters']['requestedCapacity']`.

Although the API requires slightly different data for creation than updates, Chef should be as declarative as possible. In my recipe I define it as it will exist after it's created, and let our resource figure out how to create it if it does not exist.

Hopefully that makes sense. If not, I can give an example...